src/shared/utilities/mobileDetector.ts

```diff
@@ -0,0 +1,29 @@
+/**
+ * Returns true if and only if the user is on a iOS device.
+ * Used to determine whether iOS-specific behavior should be applied.
+ */
+export const isIOS = (): boolean => {
+  if (!window || !window.navigator || !window.navigator.userAgent) {
+    return false;
+  }
+  return /iPad|iPhone|iPod/i.test(window.navigator.userAgent);
+};
+
+/**
+ * Returns true if and only if the user is on a Android device.
+ * Used to determine whether Android-specific behavior should be applied.
+ */
+export const isAndroid = (): boolean => {
+  if (!window || !window.navigator || !window.navigator.userAgent) {
+    return false;
+  }
+  return /android/i.test(window.navigator.userAgent);
+};
+
+/**
+ * Returns true if and only if the user is on a touch device and is currently using its touch API.
+ * Used to determine interaciton with hybrid touch devices like Microsoft Surface.
+ */
+export const isTouching = (): boolean => {
```

Comment:
consider a name update here. it doesnt check is user is actively "touching" but rather if the device supports "touch"... maybe `isTouchDevice` or `isTouchSupported`